### DISCUSSION TITLE
Same repo different path for flux manifest and application

### DISCUSSION THREAD

**Discussion by schandroo** [fluxcd/flux2#4101] [General]
Is it possible to use the same repository same branch with different paths for flux bootstrap and application? For example,

```
flux bootstrap git --url=<repo> --branch=master --path=flux/ ....
```
This will store the flux manifests under flux/flux-system of the repo root path.
And then create a source like this.

```
flux create source git test --url=<repo> --branch=master --path=app/deploy/ --namespace=default ....
```

Flux bootstrap succeeded for me but not the source.

**Comment by makkes** [comment]
> Flux bootstrap succeeded for me but not the source.

What is the error message?

**Reply by schandroo** [reply]
No specific error that I noticed in both the source-controller and kustomize-controller pods. Source controller pod logs even said "msg":"no changes since last reconcilation: observed".

But if I do `flux get sources git -A` only flux-system namespace source is displayed and not the new one `(test)` that I created under namespace default. 

I am newbie to flux and just learning the trade. Something more should I check?

**Reply by schandroo** [reply]
To explain further, I should say I committed the flux custom resources for source and kustomize controller under `flux/flux-system` on the root path of the repo and expected the application to be automatically deployed.
```bash
flux-system/
|-- gotk-components.yaml
|-- gotk-sync.yaml
|-- kustomization.yaml
|-- kustomizations
|   |-- kustomization.yaml
|   |-- nginxhello-kustomization.yaml
|-- sources
    |-- kustomization.yaml
    |-- nginxhello-source.yaml
```

Also created a secret and added the output key to the SSH public keys via gerrit UI.

`flux create secret git gitops-lab-deploy-key --url=<ssh://gerrit repo> --namespace default`
